quill-jdbc-monix/src/main/scala/io/getquill/context/jdbc/AsyncJdbcContext.scala

```diff
@@ -0,0 +1,248 @@
+package io.getquill.context.jdbc
+
+import scala.util.Try
+import io.getquill.NamingStrategy
+import io.getquill.context.{ Context, ContextEffect, StreamingContext }
+import io.getquill.context.sql.idiom.SqlIdiom
+import io.getquill.util.ContextLogger
+import monix.eval.Task
+import java.io.Closeable
+import java.sql.{ Array => _, _ }
+
+import javax.sql.DataSource
+import monix.execution.Scheduler
+import monix.execution.misc.Local
+import monix.reactive.Observable
+
+object Runner {
+  def default = new Runner {}
+  def using(scheduler: Scheduler) = new Runner {
+    override def schedule[T](t: Task[T]): Task[T] = t.executeOn(scheduler, true)
+    override def boundary[T](t: Task[T]): Task[T] = t.executeOn(scheduler, true)
+  }
+}
+
+trait Runner extends ContextEffect[Task] {
+  override def wrap[T](t: => T): Task[T] = Task(t)
+  override def push[A, B](result: Task[A])(f: A => B): Task[B] = result.map(f)
+  override def seq[A, B](list: List[Task[A]]): Task[List[A]] = Task.sequence(list)
+  def schedule[T](t: Task[T]): Task[T] = t
+  def boundary[T](t: Task[T]): Task[T] = t.asyncBoundary
+}
+
+/**
+ * Quill context that wraps all JDBC calls in `monix.eval.Task`.
+ *
+ */
+abstract class AsyncJdbcContext[Dialect <: SqlIdiom, Naming <: NamingStrategy](
```

Comment:
I think the `Async` prefix is misleading here, maybe `Task` or `Monix`?

Reply:
Yeah. I think `Monix` is better, especially if we're gonna have a `quill-monix` module eventually. (`Task` doesn't seem to make sense because some methods return `Observeable`). I'll refactor this.

Reply:
If we change to `MonixJdbcContext` should we have `PostgresMonixJdbcContext` or just `PostgresMonixContext`?

Reply:
I think `PostgresMonixJdbcContext` since we could have other variations if we implement monix interfaces for other modules like ndbc, finagle, etc

Reply:
Okay. Hungarian-notation will be our inspiration :)

Reply:
Done